commit 5a6b62344edd0b7f21122a46577c54aa105f8bc0 (refs/changes/15/3215/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-01-15T16:11:09+01:00 (1 year, 9 months ago)
    
    TOOLS-1973 sdcadm up moray throws an exception when a VM never made it from SAPI into VMAPI

diff --git a/lib/procedures/update-moray-v2.js b/lib/procedures/update-moray-v2.js
index bde918f..4e60481 100644
--- a/lib/procedures/update-moray-v2.js
+++ b/lib/procedures/update-moray-v2.js
@@ -12,6 +12,7 @@ var assert = require('assert-plus');
 var sprintf = require('extsprintf').sprintf;
 var util = require('util');
 var vasync = require('vasync');
+var verror = require('verror');
 
 var errors = require('../errors');
 var common = require('../common');
@@ -130,6 +131,14 @@ UpdateMorayV2.prototype.execute = function morayv2Execute(opts, cb) {
                             uuid: tmpInst.uuid
                         }, function (vmErr, vm) {
                             if (vmErr) {
+                                if (verror.hasCauseWithName(
+                                    vmErr, 'ResourceNotFoundError')) {
+                                    ctx.tmpInstToRemove =
+                                        ctx.tmpInstToRemove || [];
+                                    ctx.tmpInstToRemove.push(tmpInst.uuid);
+                                    nextVm();
+                                    return;
+                                }
                                 nextVm(new errors.SDCClientError(vmErr,
                                     'vmapi'));
                                 return;
@@ -173,7 +182,8 @@ UpdateMorayV2.prototype.execute = function morayv2Execute(opts, cb) {
                     inputs: ctx.tmpInstToRemove,
                     func: function removeFromSapi(inst, nextInst) {
                         sdcadm.sapi.deleteInstance(inst, function (remErr) {
-                            if (remErr) {
+                            if (remErr && !verror.hasCauseWithName(remErr,
+                                'ResourceNotFoundError')) {
                                 nextInst(new errors.SDCClientError(remErr,
                                     'sapi'));
                                 return;
